datastream-cloud-storage/src/main/java/com/linkedin/datastream/cloud/storage/ObjectBuilder.java

```diff
@@ -140,10 +140,10 @@ public void run() {
                 LOG.error("Unable to write to WriteLog {}", e);
                 aPackage.getAckCallback().onCompletion(new DatastreamRecordMetadata(
                         aPackage.getCheckpoint(), aPackage.getTopic(), aPackage.getPartition()), e);
-//            } catch (Exception e) {
-//                LOG.error("Unable to write to WriteLog {}", e);
-//                aPackage.getAckCallback().onCompletion(new DatastreamRecordMetadata(
-//                          aPackage.getCheckpoint(), aPackage.getTopic(), aPackage.getPartition()), e);
+            } catch (Exception e) {
+                LOG.error("Unable to write to WriteLog {}", e);
+                aPackage.getAckCallback().onCompletion(new DatastreamRecordMetadata(
+                          aPackage.getCheckpoint(), aPackage.getTopic(), aPackage.getPartition()), e);
```

Comment:
I'd let Santosh confirm if it's safe to do here.

Reply:
No. Catching a generic exception indicates we don't really understand what all possible specific exceptions can be raised raised by the implementation.

Let's please understand possible specific exceptions and handle those. Please be careful, un-handled exception can kill the object builder thread that could be servicing other streams.

Reply:
Please don't catch Exception here. Instead identify the specific exceptions can be raised and handle them.

Make sure you have full coverage of exception handling. Unhandled exception will result in dead object builder thread that could be serving other streams.